ntbs-service/Pages/Notifications/Edit/Comorbidities.cshtml.cs

```diff
@@ -18,28 +24,62 @@ public class ComorbiditiesModel : NotificationEditModelBase
         protected override async Task<IActionResult> PrepareAndDisplayPageAsync(bool isBeingSubmitted)
         {
             ComorbidityDetails = Notification.ComorbidityDetails;
+            ImmunosuppressionDetails = Notification.ImmunosuppressionDetails;
 
             await SetNotificationProperties(isBeingSubmitted, ComorbidityDetails);
+            await SetNotificationProperties(isBeingSubmitted, ImmunosuppressionDetails);
+
             if (ComorbidityDetails.ShouldValidateFull)
             {
-                TryValidateModel(ComorbidityDetails, ComorbidityDetails.GetType().Name);
+                TryValidateModel(ComorbidityDetails, nameof(ComorbidityDetails));
+            }
+
+            if (ImmunosuppressionDetails.ShouldValidateFull)
+            {
+                TryValidateModel(ImmunosuppressionDetails, nameof(ImmunosuppressionDetails));
             }
 
             return Page();
         }
 
         protected override IActionResult RedirectAfterSaveForDraft(bool isBeingSubmitted)
         {
-            return RedirectToPage("./Immunosuppression", new { NotificationId, isBeingSubmitted });
+            return RedirectToPage("./SocialContextAddresses", new { NotificationId, isBeingSubmitted });
         }
 
         protected override async Task ValidateAndSave()
         {
             ComorbidityDetails.SetFullValidation(Notification.NotificationStatus);
-            if (TryValidateModel(ComorbidityDetails, ComorbidityDetails.GetType().Name))
+            if (TryValidateModel(ComorbidityDetails, nameof(ComorbidityDetails)))
             {
                 await Service.UpdateComorbidityAsync(Notification, ComorbidityDetails);
             }
+
+            ImmunosuppressionDetails.SetFullValidation(Notification.NotificationStatus);
+            if (TryValidateModel(ImmunosuppressionDetails, nameof(ImmunosuppressionDetails)))
+            {
+                await Service.UpdateImmunosuppresionDetailsAsync(Notification, ImmunosuppressionDetails);
+            }

```

Comment:
My preference would be to only update the entities when both models have passed validation and not have it done separately